From 09cc553ede97a6e4eb442b417d80cec7a7f488ec Mon Sep 17 00:00:00 2001
From: Li Zhou <li.zhou@windriver.com>
Date: Fri, 8 Jul 2016 14:37:59 +0800
Subject: [PATCH] pci: pci-layerscape-ep: section mismatch bug fix

Fix the section mismatch warning:
WARNING: drivers/pci/host/built-in.o(.data+0xe8): Section mismatch in reference from the variable ls_pcie_ep_driver to the function .init.text:ls_pcie_ep_probe()
The variable ls_pcie_ep_driver references
the function __init ls_pcie_ep_probe()
If the reference is valid then annotate the
variable with __init* or __refdata (see linux/init.h) or name the variable:
*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console

Remove __init for function ls_pcie_ep_probe, because it is used by
ls_pcie_ep_driver, which hasn't __init* annotation.

Signed-off-by: Li Zhou <li.zhou@windriver.com>
---
 drivers/pci/host/pci-layerscape-ep.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/pci/host/pci-layerscape-ep.c b/drivers/pci/host/pci-layerscape-ep.c
index 153379b..6997e52 100644
--- a/drivers/pci/host/pci-layerscape-ep.c
+++ b/drivers/pci/host/pci-layerscape-ep.c
@@ -189,7 +189,7 @@ static int ls_pcie_ep_init(struct ls_pcie *pcie)
 	return 0;
 }
 
-static int __init ls_pcie_ep_probe(struct platform_device *pdev)
+static int ls_pcie_ep_probe(struct platform_device *pdev)
 {
 	struct ls_pcie *pcie;
 	struct resource *dbi_base;
-- 
1.7.5.4

